alibi/explainers/tests/test_simiarlity/conftest.py

```diff
@@ -1,80 +1,100 @@
+    np.random.seed(seed)
+    tf.random.set_seed(seed)
+    # Additional seeds potentially required when using a gpu
+    # (see https://www.youtube.com/watch?v=TB07_mUMt0U&t=1804s)
+    os.environ['TF_CUDNN_DETERMINISTIC'] = 'true'
+    os.environ['TF_DETERMINISTIC_OPS'] = 'true'
+    os.environ['PYTHONHASHSEED'] = str(seed)
+
+
+@pytest.fixture(params=[0, 1, 2])
+def seed(request):
+    seed = request.param
+    set_seed(seed)
+    return seed
 
 
 def get_flattened_model_parameters(model):
     """
-    Returns a flattened list of all torch or tensorflow model parameters.
+    Returns a flattened list of all `torch` or `tensorflow` model parameters.
     """
     if isinstance(model, nn.Module):
-        return np.concatenate([p.detach().numpy().reshape(-1) for p in model.parameters()])
+        return np.concatenate([_PytorchBackend.to_numpy(p).reshape(-1) for p in model.parameters()])
     elif isinstance(model, tf.keras.Model):
-        return np.concatenate([p.numpy().reshape(-1) for p in model.trainable_weights])
+        return np.concatenate([_TensorFlowBackend.to_numpy(p).reshape(-1) for p in model.trainable_weights])
 
 
 @pytest.fixture(scope='module')
 def random_reg_dataset(request):
     """
     Constructs a random regression dataset with 1d target.
     """
-    shape = request.param.get('shape', (10, ))
+    set_seed()
+    shape = request.param.get('shape', 10)
     size = request.param.get('size', 100)
-
-    # define random train set
-    x_train = np.random.randn(size, *shape).astype(np.float32)
-    y_train = np.random.randn(size, 1).astype(np.float32)
-
-    # define random test set
-    x_test = np.random.randn(size, *shape).astype(np.float32)
-    y_test = np.random.randn(size, 1).astype(np.float32)
-    return (x_train, y_train), (x_test, y_test)
+    X, Y = make_regression(n_samples=2*size, n_features=shape, n_informative=10)
+    X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.5, random_state=42)
+    Y_train, Y_test = Y_train[:, None].astype(np.float32), Y_test[:, None].astype(np.float32)
+    return (X_train, Y_train), (X_test, Y_test)
 
 
 @pytest.fixture(scope='module')
 def random_cls_dataset(request):
     """
     Constructs a random classification dataset with 10 labels.
     """
-    shape = request.param.get('shape', (10, ))
+    set_seed()
+    shape = request.param.get('shape', 10)
     size = request.param.get('size', 100)
-
-    # define random train set
-    x_train = np.random.randn(size, *shape)
-    y_train = np.random.randint(low=0, high=10, size=size).astype(np.int64)
-
-    # define random test set
-    x_test = np.random.randn(size, *shape)
-    y_test = np.random.randint(low=0, high=10, size=size).astype(np.int64)
-    return (x_train, y_train), (x_test, y_test)
+    X, Y = make_classification(n_samples=2*size, n_features=shape, n_classes=10, n_informative=10, n_redundant=0)
+    X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.5, random_state=42)
```

Comment:
Same comments re randomness as above in the regression case.

Reply:
[see above](https://github.com/SeldonIO/alibi/pull/577#discussion_r848628463)